### PR TITLE
aii-opennebula: Include OpenNebula v5.0 support

### LINKED ISSUE #187: aii-opennebula: Include OpenNebula 5.x support
ONE 5.0 includes some new features and the VM template was changed. aii-opennebula must check the ONE version through the RPC endpoint and change the VM template accordingly.

### PR DESCRIPTION
**Pull request by alvarosimon**
* Fix #187 : Include OpenNebula 5.x support

### CODE REVIEW

**Comment by stdweird** [comment]
@alvarosimon please rebase

**Comment by alvarosimon** [comment]
@stdweird  PR sync with master. Now is ready to be reviewed

File: aii-opennebula/src/main/perl/opennebula.pm [hunk 1]
```diff
@@ -86,9 +86,14 @@ sub make_one
 # Detect and process ONE templates
 sub process_template
 {
-    my ($self, $config, $tt_name) = @_;
+    my ($self, $config, $tt_name, $oneversion) = @_;
     
     my $tree = $config->getElement('/')->getTree();
+    if ((defined $oneversion) and ($oneversion >= version->new("4.9.0"))) {
+        $tree->{system}->{opennebula}->{boot} = $oneversion;
```
Comment (stdweird) [inline aii-opennebula/src/main/perl/opennebula.pm]: why don't you pass an arrayref here and do the join in the TT file?

Comment (alvarosimon) [inline aii-opennebula/src/main/perl/opennebula.pm]: I was also thinking about this. Probably in the future more things will change with another update in a different section. Maybe we can create a new function called `update_config_tree` or something like that, this function could update, depending on the version detected by the RPC, the conf tree, so we will set within this method `$tree->{system}->{opennebula}->{$section}=$version` if it's required and return the modified tree. After these changes we only use the TT file to write a specific conf for a specific version, if the version is not set we write the default setup, instead to include all the logic in the TT file. 

What do you think? I'm not sure what is the best way to handle this...

Comment (stdweird) [inline aii-opennebula/src/main/perl/opennebula.pm]: mayeb in the future yes, but for now, it's only the boot section, so i'd keep it here.

File: aii-opennebula/src/main/perl/opennebula.pm [hunk 2]
```diff
@@ -86,9 +91,16 @@ sub make_one
 # Detect and process ONE templates
 sub process_template
 {
-    my ($self, $config, $tt_name) = @_;
+    my ($self, $config, $tt_name, $oneversion) = @_;
     
     my $tree = $config->getElement('/')->getTree();
+    if ((defined $oneversion) and ($oneversion >= version->new("4.9.0"))) {
```
Comment (stdweird) [inline aii-opennebula/src/main/perl/opennebula.pm]: change version to `5.0.0` as the message mentions

File: aii-opennebula/src/main/perl/opennebula.pm [hunk 3]
```diff
@@ -86,9 +91,16 @@ sub make_one
...
+        $tree->{system}->{opennebula}->{boot} = BOOT_V5;
+        $main::this_app->info("BOOT section updated to support OpenNebula version >= 5.0.0");
+    } else {
+        $tree->{system}->{opennebula}->{boot} = BOOT_V4;
```
Comment (stdweird) [inline aii-opennebula/src/main/perl/opennebula.pm]: add message that this is boot section for older than 5.0.0

File: aii-opennebula/src/main/perl/opennebula.pm [hunk 4]
```diff
@@ -94,10 +94,11 @@ sub process_template
     my ($self, $config, $tt_name, $oneversion) = @_;
     
     my $tree = $config->getElement('/')->getTree();
-    if ((defined $oneversion) and ($oneversion >= version->new("4.9.0"))) {
+    if ((defined $oneversion) and ($oneversion >= version->new("5.0.0"))) {
         $tree->{system}->{opennebula}->{boot} = BOOT_V5;
-        $main::this_app->info("BOOT section updated to support OpenNebula version >= 5.0.0");
+        $main::this_app->info("BOOT section set to support OpenNebula versions >= 5.0.0");
```
Comment (stdweird) [inline aii-opennebula/src/main/perl/opennebula.pm]: verbose is enough

File: aii-opennebula/src/main/perl/opennebula.pm [hunk 5]
```diff
@@ -94,10 +94,11 @@ sub process_template
...
     } else {
+        $main::this_app->info("BOOT section set to support OpenNebula versions < 5.0.0");
```
Comment (stdweird) [inline aii-opennebula/src/main/perl/opennebula.pm]: verbose is enough

**Comment by alvarosimon** [comment]
@stdweird remarks addressed

**Pull request merged**